visualizer/src/service/tools/segmentMachine.js

```diff
@@ -1,37 +1,39 @@
 import { actions, assign, forwardTo, Machine, send, sendParent, spawn } from 'xstate';
-import createBrushMachine from './segment/brushMachine';
-import createFloodMachine from './segment/floodMachine';
-import createSelectMachine from './segment/selectMachine';
-import createThresholdMachine from './segment/thresholdMachine';
+import brushMachine from './segment/brushMachine';
+import floodMachine from './segment/floodMachine';
+import selectMachine from './segment/selectMachine';
+import thresholdMachine from './segment/thresholdMachine';
 import { toolActions, toolGuards } from './segment/toolUtils';
-import createTrimMachine from './segment/trimMachine';
-import createWatershedMachine from './segment/watershedMachine';
+import trimMachine from './segment/trimMachine';
+import watershedMachine from './segment/watershedMachine';
 
 const { pure, respond } = actions;
 
-// TODO: move to config file?
-const colorTools = ['brush', 'select', 'trim', 'flood'];
-
-const createToolMachineLookup = {
-  brush: createBrushMachine,
-  select: createSelectMachine,
-  threshold: createThresholdMachine,
-  trim: createTrimMachine,
-  flood: createFloodMachine,
-  watershed: createWatershedMachine,
-};
-
-const createToolMachine = context => {
-  const { tool } = context;
-  return spawn(createToolMachineLookup[tool](context), 'tool');
+const panState = {
```

Comment:
Seems like a lot of changes in this file - can you briefly summarize?

Reply:
- child tool machines no longer need the parent context to be spawned, so the imports are renamed from `createXMachine to `xMachine`
- spawning child tools is consolidated into a single entry action `spawnTools` instead of happening every time we switch tools
- new `panState` to check if we should send a `SET_PAN_ON_DRAG` event to the canvas machine
- new `syncState` that groups all the state update events to forward to child tool actors
- combine `USE_X` events into `SET_TOOL` event
- initialized the context with `null` and then syncing with events from other actors instead of hard coding multiple machines to start with the same context